scripts/cert_generator.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+usage() {
+    cat <<EOF
+Generate certificate suitable for admission webhook service.
+This script uses k8s' CertificateSigningRequest API to a generate a
+certificate signed by k8s CA suitable for admission webhook services.
+This requires permissions to create and approve CSR. See
+https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster for
+detailed explantion and additional instructions.
+The server key/cert k8s CA cert are stored in a k8s secret.
+usage: ${0} [OPTIONS]
+The following flags are required.
+      --service          Service name of webhook.
+      --namespace        Namespace where webhook service and secret reside.
+      --secret           Secret name for CA certificate and server certificate/key pair.
+EOF
+    exit 1
+}
+
+while [[ $# -gt 0 ]]; do
+    case ${1} in
+        --service)
+            service="flink-operator-webhook-service"
+            shift
+            ;;
+        --secret)
+            secret="webhook-server-cert"
+            shift
+            ;;
+        -n | --namespace)
+            namespace="flink-operator-system"
+            shift
+            ;;
+        *)
+            usage
+            ;;
+    esac
+    shift
+done
+
+[ -z ${service} ] && service="flink-operator-webhook-service"
```

Comment:
Always use "[[ ]]" instead of "[ ]".

Reply:
Because you `set -u` for this script, if `service` is not provided, `[ -z ${service} ]` would return error because you are trying to evaluate a undefined variable.